spec/unit/timeout_spec.rb

```diff
@@ -10,6 +10,17 @@
     }.to raise_error(TTY::Command::TimeoutExceeded)
   end
 
+  context "when there is no output" do
```

Comment:
This test case relies on fixture that doesn't produce any output and doesn't accept any input, hence this is an example of a script that 'hungs'. However, your test case, or rather implementation is timing out actual stdout and stderr reading which 'accidently' makes this case pass as a global timeout error.